File: 0896-31-2661-first-completely-painted-row-or-column/0896-31-2661-first-completely-painted-row-or-column.py
class Solution:
    def firstCompleteIndex(self, arr, mat):
        n, m = len(mat), len(mat[0])
        idx = [0] * (len(arr) + 1)
        
        for i in range(len(arr)):
            idx[arr[i]] = i
        
        ans = float('inf')
        
        # Check rows
        for i in range(n):
            mx = 0
            for j in range(m):
                mx = max(mx, idx[mat[i][j]])
            ans = min(ans, mx)
        
        # Check columns
        for i in range(m):
            mx = 0
            for j in range(n):
                mx = max(mx, idx[mat[j][i]])
            ans = min(ans, mx)
        
        return ans

File: 0896-31-2661-first-completely-painted-row-or-column/0896-31-2661-first-completely-painted-row-or-column.js
var firstCompleteIndex = function(arr, mat) {
    const n = mat.length, m = mat[0].length;
    const idx = new Array(arr.length + 1).fill(0);

    for (let i = 0; i < arr.length; i++) {
        idx[arr[i]] = i;
    }

    let ans = Infinity;

    for (let i = 0; i < n; i++) {
        let mx = 0;
        for (let j = 0; j < m; j++) {
            mx = Math.max(mx, idx[mat[i][j]]);
        }
        ans = Math.min(ans, mx);
    }

    for (let i = 0; i < m; i++) {
        let mx = 0;
        for (let j = 0; j < n; j++) {
            mx = Math.max(mx, idx[mat[j][i]]);
        }
        ans = Math.min(ans, mx);
    }

    return ans;
};
